'undefined reference' after autofilereloader commit

### Bug Description

Hello,
After 45363b7c3107c41f224b44184972a821345f29b2 builds no longer work against main, failing when linking the mixxx executable with the errors below.
Fresh builds (on Gentoo x86, QT 6.7.2) on prior commits run to completion.
Unfortunately I don't have the Qt skills to attempt a patch.

Many thanks for a great piece of software
Matt
```
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlapplication.cpp.o): in function `mixxx::qml::QmlApplication::~QmlApplication()':
/home/matt/mixxx/src/qml/qmlapplication.cpp:80:(.text+0x11c6): undefined reference to `vtable for AutoFileReloader'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlapplication.cpp.o): in function `AutoFileReloader::~AutoFileReloader()':
/home/matt/mixxx/src/util/autofilereloader.h:9:(.text._ZN16AutoFileReloaderD2Ev[_ZN16AutoFileReloaderD5Ev]+0x1c): undefined reference to `vtable for AutoFileReloader'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlapplication.cpp.o): in function `AutoFileReloader::~AutoFileReloader()':
/home/matt/mixxx/src/util/autofilereloader.h:9:(.text._ZN16AutoFileReloaderD0Ev[_ZN16AutoFileReloaderD5Ev]+0x1c): undefined reference to `vtable for AutoFileReloader'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlautoreload.cpp.o): in function `mixxx::qml::QmlAutoReload::QmlAutoReload()':
/home/matt/mixxx/src/qml/qmlautoreload.cpp:15:(.text+0x234c): undefined reference to `AutoFileReloader::AutoFileReloader(RuntimeLoggingCategory const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: /home/matt/mixxx/src/qml/qmlautoreload.cpp:17:(.text+0x236e): undefined reference to `AutoFileReloader::fileChanged(QString const&)'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlautoreload.cpp.o): in function `QMetaObject::Connection QObject::connect<void (AutoFileReloader::*)(QString const&), void (mixxx::qml::QmlAutoReload::*)()>(QtPrivate::FunctionPointer<void (AutoFileReloader::*)(QString const&)>::Object const*, void (AutoFileReloader::*)(QString const&), QtPrivate::ContextTypeForFunctor<void (mixxx::qml::QmlAutoReload::*)(), void>::ContextType const*, void (mixxx::qml::QmlAutoReload::*&&)(), Qt::ConnectionType)':
/usr/include/qt6/QtCore/qobject.h:263:(.text+0x23d6): undefined reference to `AutoFileReloader::staticMetaObject'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlautoreload.cpp.o): in function `non-virtual thunk to mixxx::qml::QmlAutoReload::~QmlAutoReload()':
/home/matt/mixxx/src/qml/qmlautoreload.h:11:(.text._ZN5mixxx3qml13QmlAutoReloadD2Ev[_ZN5mixxx3qml13QmlAutoReloadD5Ev]+0x2f): undefined reference to `vtable for AutoFileReloader'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlautoreload.cpp.o): in function `mixxx::qml::QmlAutoReload::~QmlAutoReload()':
/home/matt/mixxx/src/qml/qmlautoreload.h:11:(.text._ZN5mixxx3qml13QmlAutoReloadD2Ev[_ZN5mixxx3qml13QmlAutoReloadD5Ev]+0x44f): undefined reference to `vtable for AutoFileReloader'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlautoreload.cpp.o): in function `non-virtual thunk to mixxx::qml::QmlAutoReload::~QmlAutoReload()':
/home/matt/mixxx/src/qml/qmlautoreload.h:11:(.text._ZN5mixxx3qml13QmlAutoReloadD0Ev[_ZN5mixxx3qml13QmlAutoReloadD5Ev]+0x33): undefined reference to `vtable for AutoFileReloader'
/usr/lib/gcc/x86_64-pc-linux-gnu/13/../../../../x86_64-pc-linux-gnu/bin/ld: libmixxx-qml-lib.a(qmlautoreload.cpp.o): in function `mixxx::qml::QmlAutoReload::~QmlAutoReload()':
/home/matt/mixxx/src/qml/qmlautoreload.h:11:(.text._ZN5mixxx3qml13QmlAutoReloadD0Ev[_ZN5mixxx3qml13QmlAutoReloadD5Ev]+0x45f): undefined reference to `vtable for AutoFileReloader'
```

### Version

34dde6fc7a0128ecd92861b19aa063a6e7af6507

### OS

gentoo

```
src/qml/qmlautoreload.cpp:15:(.text+0x234c): undefined reference to `AutoFileReloader::AutoFileReloader(RuntimeLoggingCategory const&)
```
Looks like the signal symbol is not autogen correctly. What's strange is your are using 45363b7c3107c41f224b44184972a821345f29b2 but the QML file loader (introducing the `AutoFileReloader` changes) was only merged one commit after. Could you try with 21f4c88f6ec5e723638ba2119b290c24c7361150? Also, may I ask you to try with a clean build as well?

![image](https://github.com/user-attachments/assets/7c29a3ce-3e7f-453e-9e0c-0eed380461d5)

I think its the entire vtable
> /home/matt/mixxx/src/qml/qmlapplication.cpp:80:(.text+0x11c6): undefined reference to `vtable for AutoFileReloader'

But anyways. Pretty sure its fixed by https://github.com/mixxxdj/mixxx/commit/707a08f90316cafff215024f29b15a5ba8ed2029. @mccowanm could you do us a favor and test that since no else can currently reproduce this commit?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2792,6 +2792,8 @@ if(QML)
     # The following sources need to be in this target to get QML_ELEMENT properly interpreted
     src/control/controlmodel.cpp
     src/control/controlsortfiltermodel.cpp
+    # needed for qml/qmlautoreload.cpp
+    src/util/autofilereloader.cpp
   )
 
   # qt_finalize_target takes care that the resources :/mixxx.org/imports/Mixxx/

--- a/src/util/autofilereloader.cpp
+++ b/src/util/autofilereloader.cpp
@@ -6,6 +6,8 @@
 #include "moc_autofilereloader.cpp"
 #include "util/runtimeloggingcategory.h"
 
+AutoFileReloader::~AutoFileReloader() = default;
+
 AutoFileReloader::AutoFileReloader(const RuntimeLoggingCategory& loggingCategory)
         : m_loggingCategory(loggingCategory) {
     connect(&m_fileWatcher,

--- a/src/util/autofilereloader.h
+++ b/src/util/autofilereloader.h
@@ -10,6 +10,7 @@ class AutoFileReloader : public QObject {
     Q_OBJECT
   public:
     explicit AutoFileReloader(const RuntimeLoggingCategory& loggingCategory);
+    ~AutoFileReloader() override;
 
     void clear() {
         const auto files = m_fileWatcher.files();

